src/models/orderDao.js

```diff
@@ -30,4 +31,89 @@ const readOrderList = async (userId) => {
   return orderList;
 };
 
-module.exports = { readOrderList };
+const createOrder = async (userId, orderArr, totalPrice) => {
+  const queryRunner = appDataSource.createQueryRunner();
+
+  await queryRunner.connect();
+
+  await queryRunner.startTransaction();
+
+  try {
+    const order = await queryRunner.query(
+      `INSERT INTO orders(
+          user_id,
+          order_status_id,
+          total_price
+        ) VALUES ( ?, ?, ?)
+        `,
+      [userId, orderStatus.COMPLETED, totalPrice]
+    );
+
+    let values = [];
+    const valuesQuery = (arr, pushedArr) => {
+      arr.map((obj) => {
+        pushedArr.push(
+          `(${order.insertId}, ${
+            orderItemStatus.DELIVERYCOMPLETED
+          }, ${Object.values(obj)})`
+        );
+      });
+      return pushedArr;
+    };
+    values = valuesQuery(orderArr, values).join(',');
```

Comment:
```suggestion
    const valuesQuery = (data) => {
      return data.map((obj) => {
          return `(${order.insertId}, ${
            orderItemStatus.DELIVERYCOMPLETED
          }, ${Object.values(obj)})`
      });
    };
    const valueSet = valuesQuery(orderArr).join(',');
```

코드를 조금 더 정확히 수정해봐야 알겠지만 애초에 map을 썻으면 위에 불필요하게 빈리스트를 만들 필요가 없습니다.
넣어줄 data를 받으면 그에 맞는 value set을 만들어주기만 하면 될 것 같습니다.

Reply:
map 메소드가 리턴하는 값을 어떻게 변수에 담아야할 지 고민이었습니다. 도움 감사드립니다!